easistrain/EDD/strainStressd0cstEDD.py

```diff
@@ -0,0 +1,224 @@
+        + (
+            np.cos(delta)
+            * np.cos(theta)
+            * (
+                (np.cos(phi) * np.sin(omega))
+                - (np.cos(omega) * np.sin(phi) * np.sin(chi))
+            )
+        )
+        - np.sin(theta)
+        * ((np.cos(phi) * np.cos(omega)) + (np.sin(phi) * np.sin(chi) * np.sin(omega)))
+    )
+    q2 = (
+        np.cos(delta)
+        * np.cos(theta)
+        * ((np.cos(phi) * np.cos(omega) * np.sin(chi)) + (np.sin(phi) * np.sin(omega)))
+        - (np.cos(theta) * np.cos(phi) * np.cos(chi) * np.sin(delta))
+        - (
+            np.sin(theta)
+            * (
+                (np.cos(omega) * np.sin(phi))
+                - (np.cos(phi) * np.sin(chi) * np.sin(omega))
+            )
+        )
+    )
+    q3 = (
+        (np.cos(delta) * np.cos(theta) * np.cos(chi) * np.cos(omega))
+        + (np.cos(theta) * np.sin(delta) * np.sin(chi))
+        + (np.cos(chi) * np.sin(theta) * np.sin(omega))
+    )
+    defDirMeas = (
+        (e11 * q1 ** 2)
+        + (e22 * q2 ** 2)
+        + (e33 * q3 ** 2)
+        + (2 * e12 * q1 * q2)
+        + (2 * e13 * q1 * q3)
+        + (2 * e23 * q2 * q3)
+    )
+    return defDirMeas
+
+
+
+def deforDirMeasStress(anglesAndXEC, s11, s22, s33, s23, s13, s12):
+    phi = np.radians(anglesAndXEC[:-1, 0])
+    chi = np.radians(anglesAndXEC[:-1, 1])
+    omega = np.radians(anglesAndXEC[:-1, 2])
+    delta = np.radians(anglesAndXEC[:-1, 3])
+    theta = np.radians(0.5*anglesAndXEC[:-1, 4])
+    S1 = anglesAndXEC[-1,0]
+    dS2 = anglesAndXEC[-1,1]
+    q1 = (
+        (np.cos(theta) * np.cos(chi) * np.sin(delta) * np.sin(phi))
+        + (
+            np.cos(delta)
+            * np.cos(theta)
+            * (
+                (np.cos(phi) * np.sin(omega))
+                - (np.cos(omega) * np.sin(phi) * np.sin(chi))
+            )
+        )
+        - np.sin(theta)
+        * ((np.cos(phi) * np.cos(omega)) + (np.sin(phi) * np.sin(chi) * np.sin(omega)))
+    )
+    q2 = (
+        np.cos(delta)
+        * np.cos(theta)
+        * ((np.cos(phi) * np.cos(omega) * np.sin(chi)) + (np.sin(phi) * np.sin(omega)))
+        - (np.cos(theta) * np.cos(phi) * np.cos(chi) * np.sin(delta))
+        - (
+            np.sin(theta)
+            * (
+                (np.cos(omega) * np.sin(phi))
+                - (np.cos(phi) * np.sin(chi) * np.sin(omega))
+            )
+        )
+    )
+    q3 = (
+        (np.cos(delta) * np.cos(theta) * np.cos(chi) * np.cos(omega))
+        + (np.cos(theta) * np.sin(delta) * np.sin(chi))
+        + (np.cos(chi) * np.sin(theta) * np.sin(omega))
+    )
+    defDirMeasStress = (S1 * (s11 + s22 + s33)) + (dS2 * ((s11 * q1**2)
+    + (s22 * q2**2) + (s33 * q3**2) + (2 * s12 * q1 * q2)
+    + (2 * s13 * q1 * q3) + (2 * s23 * q2 * q3))
+    )
+    return defDirMeasStress
+
+
+def strainTensor(
+	fileRead,
+	fileSave,
+	numberOfPeaks,
+	XEC
+	):
+	
+	
+	h5Save = h5py.File(fileSave, "a")  ## create/append h5 file to save in
```

Comment:
You could use a context manager here as well `with h5py.File(...)` to avoid calling `h5Save.close()` at the end.